rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,156 @@
+            if (
+                event.address == self.steth_evm_address and
+                event.asset == A_ETH and
+                event.balance.amount == collateral_amount and
+                event.event_type == HistoryEventType.SPEND and
+                event.location_label == sender_address
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Submit {collateral_amount} ETH to Lido for receiving stETH in exchange'  # noqa: E501
+                event.counterparty = CPT_LIDO_ETH
+                #  preparing next action to be processed when erc20 transfer will be decoded by rotki  # noqa: E501
+                #  needed because submit levent is emitted prior of erc20 transfer, so it is not decoded yet  # noqa: E501
+                # TODO: to be confirmed with ROTKI team if it is not possible to have the erc20 event available before this decoder is called  # noqa: E501
+                paired_event = event
+                action_from_event_type = HistoryEventType.RECEIVE
+                action_to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                action_to_notes = f'Receive {steth_minted_tokens} stETH in exchange of the deposited ETH'  # noqa: E501
+                break
+
+        action_items = []  # also create an action item for the reception of the stETH tokens
+        if paired_event is not None and action_from_event_type is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=action_from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=A_STETH,
+                amount=steth_minted_tokens,
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_LIDO_ETH,
+                paired_event_data=(paired_event, True),
+                extra_data={'staked_eth': str(collateral_amount)},
+            ))
+        else:  # did not break/find anything
+            log.error(
+                f'At lido steth submit decoding of tx {context.transaction.tx_hash.hex()}'
+                f'did not find the decoded event of the ETH transfer',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO_ETH)
+
+    def _decode_lido_eth_staking_contract(self, context: DecoderContext) -> DecodingOutput:
+        """Decode interactions with stETH ans wstETH contracts"""
+
+        if (
+            context.tx_log.topics[0] == LIDO_STETH_SUBMITTED and
+            self.base.any_tracked([
+                hex_or_bytes_to_address(context.tx_log.topics[1]),  # sender address
+            ])
+        ):
+            return self._decode_lido_staking_in_steth(
+                context=context,
+            )
```

Comment:
request: Dont' break code in multiple lines if it fits in one

```suggestion
            context.tx_log.topics[0] == LIDO_STETH_SUBMITTED and
            (sender := self.base.any_tracked([
                hex_or_bytes_to_address(context.tx_log.topics[1]),
            ]))
        ):
            return self._decode_lido_staking_in_steth(context=context, sender=sender)
```